### ISSUE TITLE
deep guided filter

### ISSUE THREAD

**Issue by FengMu1995** [PeterL1n/RobustVideoMatting#75]
why did you set A = self.conv(torch.cat([cov_xy, var_x, base_hid], dim=1))？

**Comment by PeterL1n** [comment]
cat([cov_xy, var_x]) was the original implementation, I modified to add hidden channels as well.

**Comment by FengMu1995** [comment]
why should A be concatenated in this way，eg [cov_xy, var_x, base_hid]？

**Comment by PeterL1n** [comment]
Read `Guided Filter`, `Fast Guided Filter`, `Deep Guided Filter` papers.
